# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webbrowserpersist/nsWebBrowserPersist.h
# Commit: ef3f6fba28f9
# Full Hash: ef3f6fba28f9ce7ae1e58cb4a99231fc07f41b45
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2020-09-15 15:47:19
# Description:
#   Bug 1663499 - fix ending webbrowserpersist downloading so it only happens once, r=valentin
#   
#   This fixes a few causes of firing EndDownload twice:
#   
#   1. firing it from OnDataAvailable if the download was already canceled when the
# ==============================================================================

diff -r 8fa84be67303 -r ef3f6fba28f9 dom/webbrowserpersist/nsWebBrowserPersist.h
--- a/dom/webbrowserpersist/nsWebBrowserPersist.h	Tue Sep 15 12:33:35 2020 +0300
+++ b/dom/webbrowserpersist/nsWebBrowserPersist.h	Tue Sep 15 07:17:00 2020 +0000
@@ -168,6 +168,7 @@
   // mCancel is used from both the main thread, and (inside OnDataAvailable)
   // from a background thread.
   mozilla::Atomic<bool> mCancel;
+  bool mEndCalled;
   bool mCompleted;
   bool mStartSaving;
   bool mReplaceExisting;
